bevrand.frontend/assets/js/register.js

```diff
@@ -0,0 +1,173 @@
+            $(".notify").removeClass("active");
+            $("#notifyType").removeClass("success");
+        },2000);
+
+        return false;
+    }
+    var passwordLength = password.length;
+    if (passwordLength >= max || passwordLength < min)
+    {
+        document.getElementById("notifyType").textContent =
+            "Password length should be between "+ min +" to "+max
+        $(".notify").toggleClass("active");
+        $("#notifyType").toggleClass("success");
+
+        setTimeout(function(){
+            $(".notify").removeClass("active");
+            $("#notifyType").removeClass("success");
+        },2000);
+        return false;
+    }
+    return true;
+}
+
+function validatePasswordsAreEqual(password, checkPassword){
+    return password === checkPassword;
+}
+
+$("#successButton").click(function ()  {
+    window.location.href = '/login.html';
+});
+
+function registerUser(userList) {
+    $.ajax({
+        type: "POST",
+        url: `${config.proxyHostname}/api/register`,
+        data: userList,
+        contentType: "application/json",
+        success: function (data) {
+            username = data['username'];
+            id = data['id'];
+            $("#signUpText").value = "Welcome " + username + "!";
+            $("#registerButton").hide();
+            $("#successButton").show();
+            return
+        },
+        error: function (error) {
+            if (error.status === 400) {
+                var stringError = error.responseJSON['message'];
+                var splitError = JSON.parse(stringError.substring(stringError.indexOf('-') +1));
+
+                var alreadyExists = splitError['Error'].includes("already exists");
+
+                if(alreadyExists){
+                    var text = " already exists please pick another"
+                    var type  = splitError['Error'].split(":")[0]
+                    document.getElementById("notifyType").textContent = type + text;
+                }
+                else {
+                    document.getElementById("notifyType").textContent = splitError['Error'];
+                }
+
+                $(".notify").toggleClass("active");
+                $("#notifyType").toggleClass("success");
+
+                setTimeout(function(){
+                    $(".notify").removeClass("active");
+                    $("#notifyType").removeClass("success");
+                },4000);
+            }
+            if (error.status === 500 || error.status === 503) {
+                document.getElementById("notifyType").textContent = "Servers appear to be down"
+                $(".notify").toggleClass("active");
+                $("#notifyType").toggleClass("success");
+
+                setTimeout(function(){
+                    $(".notify").removeClass("active");
+                    $("#notifyType").removeClass("success");
+                },4000);
+            }
+
+        }
+    });
+}
+
+function mapUsertoJson(username, email, password) {
```

Comment:
Function with the same name can be found in the login.js file, perhaps they can be moved to one general method?

Reply:
Done